functions/postSurveyComplete.protected.ts

```diff
@@ -44,29 +97,28 @@ export const handler: ServerlessFunctionSignature<EnvVars, Event> = async (
       const channelAttributes = JSON.parse(channel.attributes);
 
       if (channelAttributes.surveyTaskSid) {
+        // get the postSurvey definition
+        const serviceConfig = await client.flexApi.configuration.get().fetch();
+        const { definitionVersion, hrm_base_url, hrm_api_version } = serviceConfig.attributes;
+        const postSurveyConfigJson = Runtime.getAssets()[`/formDefinitions/${definitionVersion}/insights/postSurvey.json`];
+        const hrmBaseUrl = `${hrm_base_url}/${hrm_api_version}/accounts/${serviceConfig.accountSid}`;
+
+        // get the survey task
         const surveyTask = await client.taskrouter
           .workspaces(context.TWILIO_WORKSPACE_SID)
           .tasks(channelAttributes.surveyTaskSid)
           .fetch();
 
-        const handlerPath = Runtime.getFunctions()['helpers/insightsService'].path;
-        const buildSurveyInsightsData = require(handlerPath)
-          .buildSurveyInsightsData as BuildSurveyInsightsData;
-
-        const serviceConfig = await client.flexApi.configuration.get().fetch();
-        const { definitionVersion } = serviceConfig.attributes;
-        const postSurveyConfigJson = Runtime.getAssets()[`/formDefinitions/${definitionVersion}/insights/postSurvey.json`];
-
         if (definitionVersion && postSurveyConfigJson && postSurveyConfigJson.open) {
-          const postSurveyConfigSpecs = JSON.parse(postSurveyConfigJson.open());
-  
-          const taskAttributes = JSON.parse(surveyTask.attributes);
-          const finalAttributes = buildSurveyInsightsData(postSurveyConfigSpecs)(taskAttributes, memory);
-          console.log('finalAttributes: ', JSON.stringify(finalAttributes));
+          const postSurveyConfigSpecs = JSON.parse(postSurveyConfigJson.open()) as OneToManyConfigSpecs;
 
-          await surveyTask.update({ attributes: JSON.stringify(finalAttributes) }); 
+          // parallel execution to save survey collected data in insights and hrm
+          await Promise.all([
+            saveSurveyInInsights(postSurveyConfigSpecs, memory, surveyTask),
+            saveSurveyInHRM(postSurveyConfigSpecs, memory, surveyTask, hrmBaseUrl, context.SYSTEM_SECRET),
```

Comment:
Note that it's possible that one of these succeeds and the other fails.  That then means that we have an inconsistency between the two.  When storing contacts, I don't think this is okay.  But for post-surveys, which are less important, this is fine.  But we should still eventually add post-surveys to our Insights/HRM audit checker.

Reply:
I agree with Nick here